src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -44,28 +47,90 @@ public IEnumerable<DataSource> ListAllDataSources()
 
     /// <summary>
     /// Constructor for runtimeConfig.
+    /// To be used when setting up from cli json scenario.
     /// </summary>
     /// <param name="Schema">schema.</param>
     /// <param name="DataSource">Default datasource.</param>
     /// <param name="Runtime">Runtime settings.</param>
     /// <param name="Entities">Entities</param>
+    /// <param name="DataSourceFiles">List of datasource files for multiple db scenario.</param></param>
     [JsonConstructor]
-    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities)
+    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities, DataSourceFiles? DataSourceFiles = null)
     {
         this.Schema = Schema;
         this.DataSource = DataSource;
         this.Runtime = Runtime;
         this.Entities = Entities;
-        this._dataSourceNameToDataSource = new Dictionary<string, DataSource>();
         this._defaultDataSourceName = Guid.NewGuid().ToString();
-        this._dataSourceNameToDataSource.Add(this._defaultDataSourceName, this.DataSource);
+
+        // we will set them up with default values
+        this._dataSourceNameToDataSource = new Dictionary<string, DataSource>
+        {
+            { this._defaultDataSourceName, this.DataSource }
+        };
 
         this._entityNameToDataSourceName = new Dictionary<string, string>();
+
         foreach (KeyValuePair<string, Entity> entity in Entities)
         {
             _entityNameToDataSourceName.TryAdd(entity.Key, this._defaultDataSourceName);
         }
 
+        // Multiple database scenario.
+        this.DataSourceFiles = DataSourceFiles;
+        IEnumerable<KeyValuePair<string, Entity>> allEntities = Entities.AsEnumerable();
+
+        if (DataSourceFiles is not null)
+        {
+            try
+            {
+                // Iterate through all the datasource files and load the config.
+                IFileSystem fileSystem = new FileSystem();
+                FileSystemRuntimeConfigLoader loader = new(fileSystem);
+
+                foreach (string dataSourceFile in DataSourceFiles.SourceFiles ?? Enumerable.Empty<string>())
```

Comment:
The check for nullability for DataSourceFiles.SourceFiles can be clubbed with line 83:

`if (DataSourceFiles is not null && DataSourceFiles.SourceFiles is not null)`

Reply:
fixed.